### PR TITLE
Code Quality: WCT Settings Control

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Motivation & Context

- Improved performance
- Improved maintainability
- Improved UI (UI tweaks)

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #12261
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

### CODE REVIEW

**Comment by yaira2** [comment]
> Improved performance

Can you share some numbers 👀

**Comment by d2dyno1** [comment]
I know that CommunityToolkit Labs already has a control ready for that matter [here](https://github.com/CommunityToolkit/Labs-Windows/issues/216)

**Comment by 0x5bfa** [comment]
This is experimental control and so I adopted. we can replace once that control become stable.

**Comment by 0x5bfa** [comment]
> Wasn't this supposed to be a code quality change? What types of UI changes are being introduced?

Like Windows 11 Settings controlls. full enhancing the card's border on hovering.

**Comment by Jay-o-Way** [comment]
@0x5bfa check the latest [comment](https://github.com/CommunityToolkit/Labs-Windows/issues/216#issuecomment-1654587975) from Niels about the "experimental" status. (e.g. PowerToys has been using it for quite a while now)

**Comment by 0x5bfa** [comment]
@yaira2 Ready for review

**Comment by 0x5bfa** [comment]
> @0x5bfa check the latest [comment](https://github.com/CommunityToolkit/Labs-Windows/issues/216#issuecomment-1654587975) from Niels about the "experimental" status. (e.g. PowerToys has been using it for quite a while now)

PowerToys uses them because very he introduced them. besides, we dont wanna add extra packages and its still *experimental*, but omce published in 8.0 we can switch out and remove our own.

**Review by yaira2 — CHANGES_REQUESTED** [review]
The specs for this change is to swap out the out control for the new one. I'm seeing a lot of unrelated changes here that are out of scope.

**Comment by 0x5bfa** [comment]
Just converted those ViewModels into IoC. those few lines are unrelated? i will revert again then

**Comment by yaira2** [comment]
I'll have to take a closer look

File: src/Files.App/App.xaml [hunk 1]
```diff
@@ -29,6 +29,7 @@
 			<ResourceDictionary.MergedDictionaries>
 				<XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
 				<ResourceDictionary Source="/ResourceDictionaries/PathIcons.xaml" />
+				<ResourceDictionary Source="/UserControls/Settings/SettingsControls.xaml" />
```
Comment (yaira2) [inline src/Files.App/App.xaml:32]: It might be better to merge this on a per page basis.

File: src/Files.App/Views/Settings/AboutPage.xaml [hunk 2]
```diff
@@ -3,44 +3,30 @@
...
-		<vm:AboutViewModel x:Name="ViewModel" />
-	</Page.DataContext>
-
-	<Grid>
-		<StackPanel
-			HorizontalAlignment="Stretch"
-			VerticalAlignment="Stretch"
-			Spacing="4">
+	<Grid x:Name="RootGrid">
+		<StackPanel Spacing="{StaticResource SettingsCardSpacing}">
```
Comment (yaira2) [inline src/Files.App/Views/Settings/AboutPage.xaml:15]: Is the spacing the same?

File: src/Files.App/Views/Settings/AboutPage.xaml [hunk 3]
```diff
@@ -3,44 +3,30 @@
...
 			</StackPanel.ChildrenTransitions>
 
-			<!--  Title  -->
-			<TextBlock
-				Padding="0,0,0,12"
-				FontSize="24"
-				FontWeight="Medium"
-				Text="{helpers:ResourceString Name=About}" />
+			<!--  Title (About)  -->
+			<TextBlock Style="{StaticResource SettingsHeaderTextBlockStyle}" Text="{helpers:ResourceString Name=About}" />
```
Comment (yaira2) [inline src/Files.App/Views/Settings/AboutPage.xaml:23]: Is the style the same?

File: src/Files.App/Views/Settings/AboutPage.xaml [hunk 4]
```diff
@@ -49,171 +35,127 @@
...
 
 			<!--  Open Source  -->
-			<TextBlock
-				Padding="0,16,0,4"
-				FontSize="16"
-				FontWeight="Medium"
-				Text="{helpers:ResourceString Name=OpenSource}" />
-
-			<!--  Translate  -->
-			<local:SettingsBlockControl
```
Comment (yaira2) [inline src/Files.App/Views/Settings/AboutPage.xaml:152]: Was this removed?

File: src/Files.App/Views/Settings/FoldersPage.xaml [hunk 5]
```diff
@@ -174,201 +141,179 @@
...
-
-						<!--  System Files  -->
-						<local:SettingsBlockControl Title="{helpers:ResourceString Name=ShowProtectedSystemFiles}" HorizontalAlignment="Stretch">
-							<ToggleSwitch
-								AutomationProperties.Name="{helpers:ResourceString Name=ShowProtectedSystemFiles}"
-								IsOn="{x:Bind ViewModel.ShowProtectedSystemFiles, Mode=TwoWay}"
-								Style="{StaticResource RightAlignedToggleSwitchStyle}" />
-						</local:SettingsBlockControl>
-
-						<!--  Alternate Data Streams  -->
```
Comment (yaira2) [inline src/Files.App/Views/Settings/FoldersPage.xaml:218]: Why were the comments removed?

File: src/Files.App/Views/Settings/TagsPage.xaml [hunk 6]
```diff
@@ -23,52 +25,41 @@
...
-				<local:SettingsBlockControl.ExpandableContent>
-					<StackPanel>
-						<!--  Create new tag  -->
+						<!--  Create A New Tag Section  -->
 						<Grid
 							x:Name="AddTagGrid"
 							Padding="32,12"
 							x:Load="{x:Bind ViewModel.IsCreatingNewTag, Mode=OneWay}"
-							Background="{ThemeResource CardBackgroundFillColorDefault}"
-							BorderBrush="{ThemeResource ControlElevationBorderBrush}"
```
Comment (yaira2) [inline src/Files.App/Views/Settings/TagsPage.xaml:71]: These shouldn't be changed in this PR

File: src/Files.App/Views/Settings/TagsPage.xaml [hunk 7]
```diff
@@ -166,15 +157,15 @@
...
 							MaxHeight="500"
 							Padding="12"
-							HorizontalAlignment="Stretch"
 							AllowDrop="True"
 							CanReorderItems="True"
 							IsItemClickEnabled="True"
 							ItemsSource="{x:Bind ViewModel.Tags, Mode=TwoWay}"
 							ReorderMode="Enabled"
-							SelectionMode="None">
+							SelectionMode="Single">
```
Comment (yaira2) [inline src/Files.App/Views/Settings/TagsPage.xaml:168]: Why was this changed?

**Comment by yaira2** [comment]
@0x5bfa the IoC changes look good to me.

File: src/Files.App/App.xaml.cs [hunk 8]
```diff
@@ -153,7 +153,13 @@ private IHost ConfigureHost()
 					.AddSingleton<DrivesViewModel>()
 					.AddSingleton<NetworkDrivesViewModel>()
 					.AddSingleton<OngoingTasksViewModel>()
-					.AddSingleton<AppearanceViewModel>()
+					// Settings ViewModels
```
Comment (d2dyno1) [inline src/Files.App/App.xaml.cs:156]: What benefit does it bring over just instantiating view models?

Comment (0x5bfa) [reply src/Files.App/App.xaml.cs:156]: we can use logger and services that are on that list if you add them into arguments of the constructor that injected by ioc. this is the very benefit.

Comment (0x5bfa) [reply src/Files.App/App.xaml.cs:156]: example: https://github.com/FluentHub/FluentHub/blob/main/src/FluentHub.App/ViewModels/Users/IssuesViewModel.cs#L16

File: src/Files.App/Views/Settings/AboutPage.xaml (hunk 3 above)
Comment (0x5bfa) [reply src/Files.App/Views/Settings/AboutPage.xaml:23]: Yes

File: src/Files.App/Views/Settings/AboutPage.xaml (hunk 2 above)
Comment (0x5bfa) [reply src/Files.App/Views/Settings/AboutPage.xaml:15]: Yes

File: src/Files.App/App.xaml (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/App.xaml:32]: Each settings page and properties page?

File: src/Files.App/Views/Settings/AboutPage.xaml (hunk 4 above)
Comment (0x5bfa) [reply src/Files.App/Views/Settings/AboutPage.xaml:152]: Reverted

File: src/Files.App/Views/Settings/FoldersPage.xaml (hunk 5 above)
Comment (0x5bfa) [reply src/Files.App/Views/Settings/FoldersPage.xaml:218]: Fixed

File: src/Files.App/Views/Settings/TagsPage.xaml (hunk 7 above)
Comment (0x5bfa) [reply src/Files.App/Views/Settings/TagsPage.xaml]: Fixed

File: src/Files.App/App.xaml (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/App.xaml]: Fixed

File: src/Files.App/Views/Settings/TagsPage.xaml (hunk 7 above)
Comment (yaira2) [reply src/Files.App/Views/Settings/TagsPage.xaml]: Did you have a reason to change this? I'm not against the change but I'd like to know the reasoning.

Comment (0x5bfa) [reply src/Files.App/Views/Settings/TagsPage.xaml]: Fixed

**Comment by 0x5bfa** [comment]
Ready for merging.

File: src/Files.App/App.xaml.cs (hunk 8 above)
Comment (yaira2) [reply src/Files.App/App.xaml.cs:156]: @d2dyno1

Comment (d2dyno1) [reply src/Files.App/App.xaml.cs:156]: It's weird, but I cannot mark this conversation as resolved...

File: src/Files.App/Views/Properties/CompatibilityPage.xaml
Comment (yaira2) [inline src/Files.App/Views/Properties/CompatibilityPage.xaml:1]: image
It looks like the card heights are different than what we have in `main`. I'll need some time to consider this change before deciding how to proceed.

File: src/Files.App/Views/Settings/AdvancedPage.xaml
Comment (yaira2) [inline src/Files.App/Views/Settings/AdvancedPage.xaml:1]: image
It looks like toggle switches are wrapping onto the next line when they don't need to.

File: src/Files.App/Views/Settings/TagsPage.xaml
Comment (yaira2) [inline src/Files.App/Views/Settings/TagsPage.xaml:1]: image
The "New tag" button doesn't need to wrap onto the next line.

**Comment by yaira2** [comment]
image
It appears that at a certain width, the text moves up a couple of pixels.

File: src/Files.App/Views/Settings/AdvancedPage.xaml
Comment (0x5bfa) [reply src/Files.App/Views/Settings/AdvancedPage.xaml:1]: There's threshold to wrap action controls when window is small. Don't we need this?

File: src/Files.App/Views/Properties/CompatibilityPage.xaml
Comment (0x5bfa) [reply src/Files.App/Views/Properties/CompatibilityPage.xaml:1]: They should have consistency with other settings controls. But I can apply different design template.

**Comment by 0x5bfa** [comment]
> It appears that at a certain width, the text moves up a couple of pixels.

Fixed.

File: src/Files.App/Views/Settings/AdvancedPage.xaml
Comment (0x5bfa) [reply src/Files.App/Views/Settings/AdvancedPage.xaml:1]: If you dont need this threshold I can set it 0.

Comment (yaira2) [reply src/Files.App/Views/Settings/AdvancedPage.xaml:1]: Is there a reason to keep the threshold?

**Comment by 0x5bfa** [comment]
We dont need to add our own code because WCT just released 8.0 few days ago.
Once some UI bugs fixed, I will going to open PR to update to use those controls.
